python/desc/twinkles/validate_ic.py

```diff
@@ -0,0 +1,849 @@
+                                                    spr_sys_df['twinkles_system'].iloc[0])[0]]
+            lens = db.get_lens(use_lens)
+            num_img = lens['NIMG']
+
+            mag = lens['MAG'].data[0]
+            lensed_mags = spr_sys_df['phosimMagNorm']
+            corrected_mags = []
+
+            for i in range(num_img.data[0]):
+
+                d_mags = self.get_agn_variability_mags(agn_var_params[str(agn_id.values[0])],
+                                                       lens['DELAY'].data[0][i], visit_mjd,
+                                                       spr_sys_df['redshift'].values[0])
+
+
+                test_sed = Sed()
+                test_sed.readSED_flambda('%s/%s' % (agnDir, 'agn.spec.gz'))
+                test_sed.redshiftSED(lens['ZSRC'])
+                test_f_norm = test_sed.calcFluxNorm(lensed_mags.values[i], norm_bp)
+                test_sed.multiplyFluxNorm(test_f_norm)
+                test_mag = test_sed.calcMag(bandpassDict[visit_band])
+                test_f_norm_2 = test_sed.calcFluxNorm(test_mag - d_mags[visit_band],
+                                                      bandpassDict[visit_band])
+                test_sed.multiplyFluxNorm(test_f_norm_2)
+                test_mag_2 = test_sed.calcMag(norm_bp)
+                test_mag_3 = test_mag_2 + 2.5*np.log10(np.abs(mag)[i])
+                
+                corrected_mags.append(test_mag_3)
+
+            corrected_mags = np.array(corrected_mags)
+            max_error = np.max(np.abs(corrected_mags - 
+                                      agn_var_params[str(agn_id.values[0])]
+                                      ['magNorm_static']))
+
+            if max_error > max_magNorm_err:
+                max_magNorm_err = max_error
+
+        print('------------')
+        print('AGN Image Magnitude Test Results:')
+
+        if max_magNorm_err < 0.001:
+            print('Pass: Image MagNorms are within 0.001 of correct values.')
+        else:
+            raise CatalogError('\nFail: Max image phosim MagNorm error is greater than 0.001 mags. ' + 
+                               'Max error is: %.4f mags.' % max_magNorm_err)
+
+        print('------------')
+
+        return
+
+    def get_agn_variability_mags(self, var_param_dict, time_delay, mjd, redshift):
+
+        def return_num_obj(params):
+            return 1
+
+        eg_test = egvar()
+        eg_test.num_variable_obj = return_num_obj
+
+        var_mags = eg_test.applyAgn([[0]], var_param_dict, 
+                                    mjd-time_delay,
+                                    redshift=np.array([redshift]))
+
+        filters = ['u', 'g', 'r', 'i', 'z', 'y']
+        var_mag_dict = {x:y for x,y in zip(filters, var_mags)}
+
+        return var_mag_dict
+
+        
+    def load_agn_var_params(self):
+
+        agn_var_params = {}
+
+        with open(os.path.join(os.environ['TWINKLES_DIR'], 'data', 
```

Comment:
Can you re-generate `agn_validation_params.txt`, delineating the columns with `;`?  Then, you can just split on `;` and use `json.loads()` to get the parameter dicts, rather than having to munge the strings below.